src/Compilers/VisualBasic/Portable/Binding/Binder_Query.vb

```diff
@@ -1897,7 +1896,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Dim keysSelector = keysLambdaBinder.BindExpressionRangeVariables(keys, True, groupBy,
                                                                              keysRangeVariables, diagnostics)
 
-            Dim keysLambda As New BoundQueryLambda(groupBy,
+            Dim keysLambda As New BoundQueryLambda(keysSelector.Syntax,
```

Comment:
It is not obvious what is changed here and why is the change necessary. Could you explain why keysSelector.Syntax is a better choice? BTW, what node does it refer to?